src/start_server.ts

```diff
@@ -81,44 +81,171 @@ async function applyDefaultOptions(options: ServerOptions):
+    throw new Error(e);
+  }
+}
+
+export type ServerInfo = MainlineServer | VariantServer | DispatchServer;
+
+/**
+ * The `default` or `primary` server. If only one ServerInfo is returned from
+ * startServers it must be a MainlineServer. This is the server that's running
+ * with the default configuration and not running a variant configuration.
+ */
+export interface MainlineServer {
+  kind: 'mainline';
+  server: http.Server;
+  app: express.Application;
+  options: ServerOptions;
+}
+/**
+ * These are servers which are running some named variant configuration. For
+ * multiple variant dependency directories are detected/configured , there will
+ * be one MainlineServer that serves out the default dependency directory, and
+ * one VariantServer for each other dependency directory.
+ */
+export interface VariantServer {
+  kind: 'variant';
+  server: http.Server;
+  app: express.Application;
+  options: ServerOptions;
+  variantName: string;
+}
+/**
+ * If more than one server is started by startServers, the main port will serve
+ * out a dispatch server. This server serves out an HTML interface that
+ * describes the other servers which have been started, and provides convenience
+ * links to them.
+ */
+export interface DispatchServer {
+  kind: 'dispatch';
+  server: http.Server;
+  app: express.Application;
+  options: ServerOptions;
+}
+
+/**
+ * Starts one or more web servers, based on the given options and
+ * variant bower_components directories that are found in the root dir.
+ */
+export async function startServers(options: ServerOptions):
+    Promise<ServerInfo[]> {
+  const root = options.root || process.cwd();
+  const filesInRoot = await fs.readdir(root);
+  const variantNames = filesInRoot
+                           .map(f => {
+                             const match = f.match(/^bower_components-(.*)/!);
+                             return match && match[1];
+                           })
+                           .filter(f => f != null && f !== '');
+  // TODO(rictic): support manually configuring variants? tracking more
+  //   metadata about them besides their names?
+  if (variantNames.length > 0) {
+    return await startVariants(options, variantNames);
+  }
+
+  const serverAndApp = await _startServer(options);
+  return [{
+    kind: 'mainline',
+    server: serverAndApp.server,
+    app: serverAndApp.app, options
+  }];
+}
+
+async function startVariants(options: ServerOptions, variantNames: string[]) {
+  const serverInfos: ServerInfo[] = [];
+  const mainlineOptions = Object.assign({}, options);
+  mainlineOptions.port = 0;
+  const mainServer = await _startServer(mainlineOptions);
+  serverInfos.push({
+    kind: 'mainline',
+    server: mainServer.server,
+    app: mainServer.app,
+    options: mainlineOptions,
+  });
+
+  for (const variant of variantNames) {
+    const variantOpts = Object.assign({}, options);
+    variantOpts.port = 0;
+    variantOpts.componentDir = `bower_components-${variant}`;
```

Comment:
This hardcoding seems unfortunate. How about a description of a variant that included its path.